src/VisualStudio/CodeLens/ReferenceCodeLensProvider.cs

```diff
@@ -73,37 +73,72 @@ public ReferenceCodeLensProvider(Lazy<ICodeLensCallbackService> codeLensCallback
             return SpecializedTasks.False;
         }
 
-        public async Task<IAsyncCodeLensDataPoint> CreateDataPointAsync(
+        public Task<IAsyncCodeLensDataPoint> CreateDataPointAsync(
             CodeLensDescriptor descriptor, CodeLensDescriptorContext descriptorContext, CancellationToken cancellationToken)
         {
             var dataPoint = new DataPoint(
                 this,
                 _lazyCodeLensCallbackService.Value,
-                descriptor,
-                await TryGetConnectionAsync(cancellationToken).ConfigureAwait(false));
+                descriptor);
 
-            await dataPoint.TrackChangesAsync(cancellationToken).ConfigureAwait(false);
-
-            return dataPoint;
+            AddDataPoint(dataPoint);
+            return Task.FromResult<IAsyncCodeLensDataPoint>(dataPoint);
         }
 
-        private async Task<Stream?> TryGetConnectionAsync(CancellationToken cancellationToken)
+        private async Task PollForUpdatesAsync()
```

Comment:
📝 The current CodeLens API does not support a callback invalidation object, so prior to the transition of this feature to LSP we have to poll for updates.

Reply:
Worth adding a comment on the design here so we know that it can be changed with the LSP transition

Reply:
Done